crates/ra_hir/src/ty.rs

```diff
@@ -267,8 +251,9 @@ impl Ty {
     /// `Option<u32>` afterwards.)
     pub fn apply_substs(self, substs: Substs) -> Ty {
         match self {
-            Ty::Adt { def_id, .. } => Ty::Adt { def_id, substs },
-            Ty::FnDef { def, .. } => Ty::FnDef { def, substs },
+            Ty::Apply(ApplicationTy { name, .. }) => {
+                Ty::Apply(ApplicationTy { name, parameters: substs })
```

Comment:
should we check that before/after subst lengths are the same?

Reply:
Yep, and I found a bug when I did it ;)